### ISSUE TITLE
[FEATURE] Stonecutter Implementation

### ISSUE THREAD

**Issue by ginsm** [ginsm/forgotten-graves#93]
**TL;DR:** I'll be focusing on a large codebase rework that should make maintaining multiple Minecraft versions much easier. It will take time and I may neglect other features/bug fixes whilst I work on it.

----

I've been considering using a multi-version repository for a while now. It seems like [stonecutter-kt](https://github.com/kikugie/stonecutter-kt) would help with that. This should probably be my priority for now as it would make maintaining different versions much easier.

I'm considering creating classes that wrap Minecraft and Fabric methods that are often used throughout the codebase. The wrapper methods would utilize stonecutter to modify the code for each version, which  would result in the wrapper method essentially becoming version agnostic.

For example, imagine seeing this throughout the codebase for every translatable text call:

```java
/*? if >1.19 {*/
return Text.translatable(...)
/*?} else {*//*
return new TranslatableText(...)
*//*?}*/
```

It would get very cluttered very quickly. And if they changed to another syntax in 1.21 or something? Yikes. But if I write a wrapper method with that same code, the rest of the codebase can simply use the method and it would be much easier to read (and much easier to maintain, as I can use it without thinking about version-specific code).

This change would also allow me to clean up the repository.. I could remove all of the version branches and use branches with something like a git-flow methodology where branches represent a specific feature, release, hotfix, etc.

**Comment by ginsm** [comment]
Finally got around to actually setting up stonecutter-kt. Next I'll be going through each version, one-by-one, porting each to the multi-version codebase and decoupling version-specific code to the abstraction (wrapper) classes (where most of the stonecutter code will reside): [FabricAbstractions](https://github.com/ginsm/forgotten-graves/blob/feature/stonecutter/src/main/java/me/mgin/graves/abstraction/FabricAbstraction.java), [MinecraftAbstractions](https://github.com/ginsm/forgotten-graves/blob/feature/stonecutter/src/main/java/me/mgin/graves/abstraction/MinecraftAbstraction.java).

**Comment by ginsm** [comment]
I'll be porting `1.20` and `1.20.1` next and then start porting to `1.20.5` and `1.20.6`.. and then working on that GitHub Actions workflow so I can publish multiple versions at once.

**Comment by ginsm** [comment]
Alright, 1.20 and 1.20.1 are ported over. I need to test the currently ported versions more thoroughly and then I'll start porting `1.20.5` and `1.20.6`.

**Comment by ginsm** [comment]
The GitHub Actions workflow file has been created and tested; seems to be working just fine. In addition, I've archived the supported branches (see tags) and the `feature/stonecutter` branch.

I'll be starting on porting to 1.20.5 and 1.20.6 soon.
